components/Form/FormInput.js

```diff
@@ -0,0 +1,53 @@
+import classNames from 'classnames';
+import { useState } from 'react';
+
+const specialChars = /[`!@#$%^&*()_+\-=[\]{};':"\\|,.<>/?~]/;
+
+// const regex = /[a-zA-Z0-9 .]+/;
+
+export const FormInput = ({
+  defaultValue,
+  onChange,
+  labelClassName,
+  label,
+  required,
+  value,
+  pattern,
+  className,
+  ...props
+}) => {
+  const [userInput, setUserInput] = useState(defaultValue);
+  const [isValid, setIsValid] = useState(true);
+
+  const inputHandler = (e) => {
+    if (onChange) {
+      onChange(e);
+    } else {
+      pattern && setIsValid(!specialChars.test(e.target.value));
+      setUserInput(e.target.value);
+    }
+  };
+
+  return (
+    <>
+      {label && (
+        <label className={labelClassName} htmlFor={props.id}>
+          {required ? `${label} *` : label}
+          {!isValid && (
+            <span className="ml-3 text-red-600">Obsahuje nepovolené znaky</span>
+          )}
```

Comment:
Good job. You went really far with the inputs! Just embrace this moment. It is probably the last time you do it manually. 

Usually, you would use the library for it - I did want to do it the slower/painful/manual way to learn. Just for you to know, there are 3 libraries that anyone would consider using - Formik, FinalForm and react-hook-form

Not sure anybody uses something else. If somebody uses redux-forms, they have not idea what they are doing